piqe_ocp_lib/api/resources/ocp_secrets.py

```diff
@@ -61,6 +57,48 @@ def get_secret_token(self, secret_name, namespace="default"):
 
         return secret_token
 
+    def get_secret_names(self, namespace="default"):
+        """
+        Get service account secret token from openshift cluster
+        :param namespace: (string) name of namespace where secret exist
+        :return: (string) secret token for specified secret
+        """
+        api_response = None
+        secret_name_list = list()
+        try:
+            api_response = self.ocp_secret.get(namespace=namespace)
+        except ApiException as e:
+            logger.exception("Exception while getting service account : %s\n" % e)
+
+        if api_response:
+            for secret in api_response.items:
+                secret_name_list.append(secret["metadata"]["name"])
+
+        return secret_name_list
+
+    def get_long_live_bearer_token(self, sub_string="default-token", namespace="default"):
+        """
+        Get bearer token from secrets to authorize openshift cluster
+        :param sub_string: (str) substring of secrets name to find actual secret name since openshift append random
+        5 ascii digit at the end of every secret name
+        :param namespace: (string) name of namespace where secret exist
+        :return: (string) secret token for specified secret
+        """
+        secret_name_list = self.get_secret_names(namespace=namespace)
+        try:
+            secret_name = next(name for name in secret_name_list if sub_string in name)
+        except StopIteration as e:
```

Comment:
Moving forward we should raise the exception so the calling method can make a determination on whether to act or not.

Reply:
Sure